src/main/webapp/script.js

```diff
@@ -11,3 +11,72 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+/**
+ * Fetches questions from server, wraps each in an <li> element, 
+ * and adds them to the DOM.
+ */
+async function fetchForum() {
+  const response = await fetch('/fetch-forum');
+  const questionsObject = await response.json();
+  const questionsContainer = document.getElementById('forum');
+  questionsContainer.innerHTML = '';
+  questionsObject.forEach(question => {
+    questionsContainer.appendChild(createQuestionElement(question));
+  });
+}
+
+/** 
+ * Creates an <li> element with question data. 
+ * Each element corresponds to a question to be displayed in the DOM.
+ */
+function createQuestionElement(question) {
+  const questionElement = document.createElement('li');
+  questionElement.setAttribute('class', 'list-group-item');
+  questionElement.innerText = question.title;
+  
+  // Asker name is placed besides the question.
+  const askerElement = document.createElement('small');
+  askerElement.setAttribute('class', 'text-muted');
+  askerElement.innerText = '\t' + question.askerName;
+  questionElement.appendChild(askerElement);
+
+  // Number of followers is placed to the right side at the top.
+  const followersElement = document.createElement('small');
+  followersElement.setAttribute('class', 'float-right');
+  if (question.numberOfFollowers === 1) {
+    // Avoid writing '1 followers'.
+    followersElement.innerText = question.numberOfFollowers + ' follower';
+  } else {
+    followersElement.innerText = question.numberOfFollowers + ' followers';
+  }
+  questionElement.appendChild(followersElement);
+
+  // Number of answers is placed to the right side at the bottom.
+  const answersElement = document.createElement('small');
+  answersElement.setAttribute('class', 'float-right');
+  if (question.numberOfAnswers === 1) {
+    // Avoid writing '1 answers'.
+    answersElement.innerText = question.numberOfAnswers + ' answer';
+  } else {
+    answersElement.innerText = question.numberOfAnswers + ' answers';
+  }
+  questionElement.appendChild(document.createElement('br'));
```

Comment:
Tim and Gloria recommend not adding line breaks to the HTML, but instead use CSS. I'm not requesting a change here, just keep in mind they might.

Reply:
i might look into it later, thanks!